Sources/CodeEditTextView/TreeSitter/TreeSitterClient+Highlight.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  TreeSitterClient+Highlight.swift
+//  
+//
+//  Created by Khan Winter on 3/10/23.
+//
+
+import Foundation
+import SwiftTreeSitter
+import CodeEditLanguages
+
+extension TreeSitterClient {
+
+    /// Queries the given language layer for any highlights.
+    /// - Parameters:
+    ///   - layer: The layer to query.
+    ///   - textView: A text view to use for contextual data.
+    ///   - range: The range to query for.
+    /// - Returns: Any ranges to highlight.
+    internal func queryLayerHighlights(layer: LanguageLayer,
+                                       textView: HighlighterTextView,
+                                       range: NSRange) -> [HighlightRange] {
```

Comment:
### [CodeStyle]:

Maybe try to not indent to much. In the `CodeEdit` repo this is not allowed by SwiftLint. We should keep it consistent.

```suggestion
    internal func queryLayerHighlights(
        layer: LanguageLayer,
        textView: HighlighterTextView,
        range: NSRange
    ) -> [HighlightRange] {
```